src/core/Akka/Event/DeadLetterListener.cs

```diff
@@ -49,35 +51,235 @@ protected override void PostStop()
             _eventStream.Unsubscribe(Self);
         }
 
+        private void IncrementCount()
+        {
+            if (_count == int.MaxValue)
+            {
+                Logging.GetLogger(Context.System, this).Info("Resetting DeadLetterListener counter after reaching Int.MaxValue.");
+                _count = 1;
+            }
+            else
+            {
+                _count++;
+            }
+        }
+
         /// <summary>
         /// TBD
         /// </summary>
         /// <param name="message">TBD</param>
         /// <returns>TBD</returns>
         protected override bool Receive(object message)
         {
-            var deadLetter = (DeadLetter)message;
-            var snd = deadLetter.Sender;
-            var rcp = deadLetter.Recipient;
+            if (_isAlwaysLoggingDeadLetters)
+            {
+                return ReceiveWithAlwaysLogging()(message);
```

Comment:
This is the correct way of calling a Receive delegate, right?

Reply:
Rather than return a `Receive` delegate here, I'd probably just feed the message directly into the handler. Fewer allocations this way.

Reply:
But then I need to pass a Receive delegate to the `Context.Become`. See line [133](https://github.com/akkadotnet/akka.net/blob/59fe3556ef1cf07fd50b32b39b92a1a849eb096e/src/core/Akka/Event/DeadLetterListener.cs#L133) for instance.

Reply:
Got it - that makes sense. I'm just worried about the memory footprint of returning new delegates for an actor that lives forever and might be receiving `DeadLetter`s regularly in a busy system. What would be a reasonable way of testing that? I just want to make sure that nothing gets rooted inside the `delegate`s so they can be GCed regularly.

Reply:
I'll look into it. Do you have an alternative for this scenario, anyway?

Reply:
Eh, there probably isn't one. We should just merge this in then.